internal/batches/util/repo.go

```diff
@@ -1,16 +1,40 @@
 package util
 
 import (
-	"github.com/sourcegraph/sourcegraph/lib/batches/template"
+	"crypto/sha256"
+	"encoding/base64"
+	"strings"
 
-	"github.com/sourcegraph/src-cli/internal/batches/graphql"
+	"github.com/sourcegraph/sourcegraph/lib/batches/template"
 )
 
-// GraphQLRepoToTemplatingRepo transforms a given *graphql.Repository into a
+// NewTemplatingRepo transforms a given *graphql.Repository into a
 // template.Repository.
-func GraphQLRepoToTemplatingRepo(r *graphql.Repository) template.Repository {
+func NewTemplatingRepo(repoName string, fileMatches map[string]bool) template.Repository {
 	return template.Repository{
-		Name:        r.Name,
-		FileMatches: r.FileMatches,
+		Name:        repoName,
+		FileMatches: fileMatches,
+	}
+}
+
+func SlugForPathInRepo(repoName, commit, path string) string {
```

Comment:
Random thought: instead of a `util` package, could we have a `repository` package? `batches/repository.SlugForPath` is a good name.